pallas-txbuilder/src/asset.rs

```diff
@@ -0,0 +1,54 @@
+use std::marker::PhantomData;
+
+use indexmap::IndexMap;
+use pallas_codec::utils::{Bytes, KeyValuePairs};
+use pallas_primitives::babbage::PolicyId;
+
+#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
+pub enum AssetError {
+    #[error("Invalid asset name")]
+    InvalidAssetName(String),
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct MultiAsset<T> {
```

Comment:
Maybe rename this `MultiAssetBuilder` just not to collide with `pallas_primitives::babbage::Multiasset<T>`